src/array/cuda/dgl_cub.cuh

```diff
@@ -7,13 +7,11 @@
 #ifndef DGL_ARRAY_CUDA_DGL_CUB_CUH_
 #define DGL_ARRAY_CUDA_DGL_CUB_CUH_
 
-// include cub in a safe manner
-#define CUB_NS_PREFIX namespace dgl {
-#define CUB_NS_POSTFIX }
-#define CUB_NS_QUALIFIER ::dgl::cub
+// This should be defined in CMakeLists.txt
+#ifndef THRUST_CUB_WRAPPED_NAMESPACE
```

Comment:
This macro has been introduced since CUB/Thrust 1.14.0. Should we update the external CUB/Thrust version for CUDA <= 11.0, though it works well now since the minimal Pytorch version we are supporting is 1.9.0, in which they wrapped CUB in its own namespace?

Reply:
Do you know if cub 1.14.0 works with cuda 10.2? I think that would be the only blocker if it does not.

Reply:
From https://nvidia.github.io/thrust/setup/requirements.html, it says
> Releases of Thrust and CUB are only tested against the latest releases of NVHPC and CUDA. It may be possible to use newer version of Thrust and CUB with an older NVHPC or CUDA installation by using a Thrust and CUB release from GitHub, but please be aware this is not officially supported.

Reply:
I tried using `cub:DeviceSegmentedSort` from CUB 1.17 with CUDA 11.7, and got an error from compute-sanitizer
```
Program hit CUDA_ERROR_NOT_FOUND (error 500) due to "named symbol not found" on CUDA API call to cuGetProcAddress.
```
while `cub:DeviceSegmentedRadixSort` works well. So I temporarily abandon upgrading CUB/Thrust.